tests/ui/transmutability/uninhabited.rs

```diff
@@ -16,6 +16,11 @@ mod assert {
             }
         }>
     {}
+
+    pub fn is_transmutable<Src, Dst>()
+    where
+            Dst: TransmuteFrom<Src>, // safety is NOT assumed, but proven
+    {}
```

Comment:
```suggestion
    pub fn is_valid<Src, Dst>()
    where
            Dst: TransmuteFrom<Src, { Assume::SAFETY} >,
    {}
```

`Assume::safety` (name needs bikeshedding, I know) is a catch-all for other safety properties not further delineated by `Assume`. In particularly, it covers library safety invariants on fields. Since I don't think you're trying to test that we're checking library safety invariants on fields (which is what's currently triggering the selection failure, per the error message), you should probably `Assume::SAFETY`.

Reply:
I copied this verbatim from the testcase that used to ICE.

Reply:
It seems to ICE with and without `Assume::SAFETY`, but it seems better to stay as close as possible to the original testcase? I have no idea how this relates to the underlying code though. I can also change it if you think that's the better test. :shrug:

Reply:
It seems to ICE even with the is_maybe_transmutable that's already in this test, so I will just use that.

Reply:
Yeah, `is_maybe_transmutable` makes for a better test, as it has the fewest early bailouts.